bot/exts/evergreen/tic_tac_toe.py

```diff
@@ -0,0 +1,322 @@
+            4: Emojis.number_emojis[4],
+            5: Emojis.number_emojis[5],
+            6: Emojis.number_emojis[6],
+            7: Emojis.number_emojis[7],
+            8: Emojis.number_emojis[8],
+            9: Emojis.number_emojis[9]
+        }
+
+        self.current = self.players[0]
+        self.next = self.players[1]
+
+        self.winner: t.Optional[t.Union[Player, AI]] = None
+        self.loser: t.Optional[t.Union[Player, AI]] = None
+        self.over = False
+        self.canceled = False
+        self.draw = False
+
+    async def get_confirmation(self) -> t.Tuple[bool, t.Optional[str]]:
+        """
+        Ask does user want to play TicTacToe against requester. First player is always requester.
+
+        This return tuple that have:
+        - first element boolean (is game accepted?)
+        - (optional, only when first element is False, otherwise None) reason for declining.
+        """
+        confirm_message = await self.ctx.send(
+            CONFIRMATION_MESSAGE.format(
+                opponent=self.players[1].user.mention,
+                requester=self.players[0].user.mention
+            )
+        )
+        await confirm_message.add_reaction(Emojis.confirmation)
+        await confirm_message.add_reaction(Emojis.decline)
+
+        def confirm_check(reaction: discord.Reaction, user: discord.User) -> bool:
+            """Check is user who reacted from who this was requested, message is confirmation and emoji is valid."""
+            return (
+                reaction.emoji in (Emojis.confirmation, Emojis.decline)
+                and reaction.message.id == confirm_message.id
+                and user == self.players[1].user
+            )
+
+        try:
+            reaction, user = await self.ctx.bot.wait_for(
+                "reaction_add",
+                timeout=60.0,
+                check=confirm_check
+            )
+        except asyncio.TimeoutError:
+            self.over = True
+            self.canceled = True
+            await confirm_message.delete()
+            return False, "Running out of time... Cancelled game."
+
+        await confirm_message.delete()
+        if reaction.emoji == Emojis.confirmation:
+            return True, None
+        else:
+            self.over = True
+            self.canceled = True
+            return False, "User declined"
+
+    async def add_reactions(self, msg: discord.Message) -> None:
+        """Add number emojis to message."""
+        for nr in Emojis.number_emojis.values():
+            await msg.add_reaction(nr)
+
+    def format_board(self) -> str:
+        """Get formatted tic-tac-toe board for message."""
+        board = list(self.board.values())
+        return "\n".join(
+            (f"{board[line]} {board[line + 1]} {board[line + 2]}" for line in range(0, len(board), 3))
+        )
+
+    async def play(self) -> None:
+        """Start and handle game."""
+        await self.ctx.send("It's time for game! Let's begin.")
```

Comment:
Maybe `time for a game` / `time for the game` reads better here?

Reply:
Not sure about it. I don't know these articles stuff well.